lib/codegen/fromcto/csharp/csharpvisitor.js

```diff
@@ -235,6 +236,39 @@ class CSharpVisitor {
         return null;
     }
 
+    /**
+     * Get dotnet namespace of a given type if it belongs to different namespace from associated concept/class
+     * @param {string} type name of the super type
+     * @param {ClassDeclaration} classDeclaration of the class
+     * @param {Object} parameters - the parameter
+     * @returns {string} the dotnet namespace of given type
+     * @private
+     */
+    getDotNetNamespaceOfType(type, classDeclaration, parameters) {
+        let dotnetNs = '';
+        // Resolve to dotnet namespace only if its a non system type
+        if (type &&  ModelUtil.parseNamespace(ModelUtil.getNamespace(type)).name !== 'concerto') {
+            const mm = classDeclaration.getModelFile()?.getModelManager();
+            let typeNamespace = mm?.getType(type)?.getNamespace();
+            if (typeNamespace && typeNamespace !== classDeclaration.getNamespace()) {
+                // Ensure non-empty string, that is separated by a period
+                let namespacePrefix = parameters.namespacePrefix ? parameters.namespacePrefix : '';
+                if (namespacePrefix !== '' && namespacePrefix.slice(-1) !== '.') {
+                    namespacePrefix += '.';
+                }
+                const otherModelFile = mm?.getModelFile(typeNamespace);
+                if (!otherModelFile) {
+                    // Couldn't resolve the other model file.
+                    dotnetNs = `${namespacePrefix}${typeNamespace}`;
```

Comment:
Can you add a test case for this line?

Reply:
Looks like we don't need this check. I added it in defense earlier, inspired by https://github.com/accordproject/concerto-codegen/blob/main/lib/codegen/fromcto/csharp/csharpvisitor.js#L129. Now we are checking if the ns is available in the model manager before reaching this check.